react/components/SearchContents/index.js

```diff
@@ -0,0 +1,118 @@
+  }
+
+  static defaultProps = {
+    type: null,
+  }
+
+  state = {
+    page: 1,
+    per: 12,
+    hasMore: true,
+  }
+
+  loadMore = fetchMore => () => {
+    const { page, per } = this.state;
+
+    fetchMore({
+      variables: { page: page + 1, per },
+      updateQuery: (prevResult, { fetchMoreResult }) => ({
+        contents: [
+          ...prevResult.contents,
+          ...fetchMoreResult.contents,
+        ],
+      }),
+    }).then(({ errors, data }) => {
+      const { contents: { length } } = data;
+      const hasMore = !errors && length > 0 && length >= per;
+
+      this.setState({
+        page: page + 1,
+        hasMore,
+      });
+    });
+  }
+
+  render() {
+    const { per, hasMore } = this.state;
+    const { type, fetchPolicy, q } = this.props;
+
+    return (
+      <Query
+        query={searchContentsQuery}
+        variables={{
+          type, per, q,
+        }}
+        fetchPolicy={fetchPolicy}
+      >
+        {({
+          loading, error, data, fetchMore,
+        }) => {
+          if (error) {
+            return (
+              <ErrorAlert>
+                {error.message}
+              </ErrorAlert>
+            );
+          }
+
+          if (!data.contents) {
+            return <BlocksLoadingIndicator />;
+          }
+
+          const { contents } = data;
+
+          return (
+            <div>
+              {loading &&
+                <BlocksLoadingIndicator />
+              }
+
+              {!loading && contents.length === 0 &&
+                <SearchEmptyMessage term={q} type={type} />
+              }
+
+              {!loading && contents.length > 0 &&
+                <Grid
+                  pageStart={1}
+                  threshold={800}
+                  initialLoad={false}
+                  loader={<BlocksLoadingIndicator key="loading" />}
+                  hasMore={contents.length >= per && hasMore}
+                  loadMore={this.loadMore(fetchMore)}
+                >
+                  {contents.map(cell => ({
+                    Image: <Cell.Konnectable key={`${cell.__typename}_${cell.id}`} konnectable={cell} />,
+                    Attachment: <Cell.Konnectable key={`${cell.__typename}_${cell.id}`} konnectable={cell} />,
+                    Text: <Cell.Konnectable key={`${cell.__typename}_${cell.id}`} konnectable={cell} />,
+                    Link: <Cell.Konnectable key={`${cell.__typename}_${cell.id}`} konnectable={cell} />,
+                    Embed: <Cell.Konnectable key={`${cell.__typename}_${cell.id}`} konnectable={cell} />,
+                    Channel: <Cell.Konnectable key={`${cell.__typename}_${cell.id}`} konnectable={cell} />,
+                    User: <Cell.Identifiable key={`${cell.__typename}_${cell.id}`} identifiable={cell} />,
+                    Group: <Cell.Identifiable key={`${cell.__typename}_${cell.id}`} identifiable={cell} />,
+                  }[cell.__typename]))}
```

Comment:
So you probably want to wrap these in functions, like https://github.com/aredotna/ervell/blob/master/react/components/Cell/components/Konnectable/index.js#L120-L128

This is rendering all of these at once, currently

Reply:
Shall do